client/signup/steps/about/index.jsx

```diff
@@ -199,18 +198,17 @@ class AboutStep extends Component {
 			eventAttributes.site_title = siteTitleInput || 'N/A';
 		}
 
-		//Site Topic
-		const englishSiteTopicInput = this.state.hasPrepopulatedVertical
+		// Set Site Topic value for tracking/marketing
+		eventAttributes.site_topic = this.state.hasPrepopulatedVertical
 			? this.state.siteTopicValue
-			: findKey( hints, siteTopic => siteTopic === siteTopicInput ) || siteTopicInput;
+			: this.state.siteTopicSlug || siteTopicInput || 'N/A';
 
-		eventAttributes.site_topic = englishSiteTopicInput || 'N/A';
 		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_topic', {
 			value: eventAttributes.site_topic,
 		} );
 
 		this.props.setSurvey( {
-			vertical: englishSiteTopicInput,
+			vertical: eventAttributes.site_topic,
```

Comment:
If the site topic field is empty, then hitting the Back button from a later step will fill the site topic field with `N/A`.

Reply:
I like your 👀